tests/Integration/ApiPlatform/DiscountEndpointTest.php

```diff
@@ -0,0 +1,180 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PsApiResourcesTest\Integration\ApiPlatform;
+
+use PrestaShop\PrestaShop\Core\Domain\Discount\Query\GetDiscountForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Discount\Command\AddDiscountCommand;
+use PrestaShop\PrestaShop\Core\Domain\Discount\ValueObject\DiscountType;
+use Tests\Resources\Resetter\LanguageResetter;
+use Tests\Resources\Resetter\DiscountResetter;
+use Tests\Resources\ResourceResetter;
+
+class DiscountEndpointTest extends ApiTestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        parent::setUpBeforeClass();
+        (new ResourceResetter())->backupTestModules();
+        DiscountResetter::resetProducts();
+        LanguageResetter::resetLanguages();
+        self::addLanguageByLocale('fr-FR');
+        self::createApiClient(['discount_write', 'discount_read']);
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        parent::tearDownAfterClass();
+        DiscountResetter::resetProducts();
+        LanguageResetter::resetLanguages();
+        // Reset modules folder that are removed with the FR language
+        (new ResourceResetter())->resetTestModules();
+    }
+
+    /**
+     * @depends testAddDiscount
+     *
+     * @param int $discountId
+     * @return int
+     */
+    public function testGetDiscount(int $discountId): int
+    {
+        // skip test if class does not exist
+        if (!class_exists(GetDiscountForEditing::class)) {
+            $this->markTestSkipped('GetDiscountForEditing class does not exist');
+        }
+
+        $bearerToken = $this->getBearerToken(['discount_read']);
+        $response = static::createClient()->request('GET', '/discount/' . $discountId, [
+            'auth_bearer' => $bearerToken,
+        ]);
+        self::assertResponseStatusCodeSame(200);
+
+        $decodedResponse = json_decode($response->getContent(), true);
+        $this->assertNotFalse($decodedResponse);
+        $this->assertEquals(
+            [
+            ],
+            $decodedResponse
+        );
+
+        return $discountId;
+    }
+
+
+    /**
+     * @dataProvider discountTypesDataProvider
+     *
+     * @param string $type
+     * @param array $names
+     * @return int
+     */
+    public function testAddDiscount(string $type, array $names): int
+    {
+        // skip test if class does not exist
+        if (!class_exists(AddDiscountCommand::class)) {
```

Comment:
Ah, I was searching how to do this, perfect!

I'm just curious, there's no need to `return;` after this method is called?

Reply:
Aaah it actually throws an exception under the hood, so no need for a return